// Copyright (c) 2023 dingodb.com, Inc. All Rights Reserved
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "common.proto";
import "error.proto";

package dingodb.pb.fileservice;

option java_package = "io.dingodb.fileservice";
option cc_generic_services = true;

message GetFileRequest {
  dingodb.pb.common.RequestInfo request_info = 1;
  int64 reader_id = 2;
  string filename = 3;
  int64 offset = 4;
  int64 size = 5;
}

message GetFileResponse {
  dingodb.pb.common.ResponseInfo response_info = 1;
  dingodb.pb.error.Error error = 2;
  // Data is in attachment
  bool eof = 3;
  int64 read_size = 4;
}

message CleanFileReaderRequest {
  dingodb.pb.common.RequestInfo request_info = 1;
  int64 reader_id = 2;
}

message CleanFileReaderResponse {
  dingodb.pb.common.ResponseInfo response_info = 1;
  dingodb.pb.error.Error error = 2;
}

service FileService {
  rpc GetFile(GetFileRequest) returns (GetFileResponse);
  // Clean file reader
  rpc CleanFileReader(CleanFileReaderRequest) returns (CleanFileReaderResponse);
}
